### PR TITLE
deprecate: manual ttid widget

### LINKED ISSUE #2449: Deprecate manual TTID / `SentryDisplayWidget`
## Current state

We currently allow two options:
- automatic ttid which calls `addPostFrameCallback` within `didPush` of the navigator observer
- manual ttid which calls `addPostFrameCallback` within a widget's `initState` that you need to wrap around the widget you want to measure from

## Why deprecate the manual version?
After some tests I've come to the conclusion that the automatic version is more accurate than we think (see timeline), there is usually a difference of around 2-5ms (give or take) to the actual frame duration which is negligible. 

By deprecating it / removing it in v9 we reduce complexity of the implementation and usage but still report accurate data.
Image

### PR DESCRIPTION
**Pull request by buenaflor**
Closes https://github.com/getsentry/sentry-dart/issues/2449

### CODE REVIEW

File: flutter/test/navigation/time_to_display_tracker_test.dart [hunk 1]
```diff
@@ -109,7 +110,7 @@ void main() {
 
         final transaction = fixture.getTransaction() as SentryTracer;
         await sut.track(transaction, startTimestamp: fixture.startTimestamp);
-      });
+      }, skip: true);
```
Comment (krystofwoldrich) [inline flutter/test/navigation/time_to_display_tracker_test.dart]: I think this should be a separate PR.

**Review by krystofwoldrich — APPROVED** [review]
One nit, otherwise looks good.

File: flutter/test/navigation/time_to_display_tracker_test.dart (hunk 1 above)
Comment (buenaflor) [reply flutter/test/navigation/time_to_display_tracker_test.dart]: the test is a manual TTID test, related but let's do it consistently, I'll open another PR
